Sources/Sentry/Public/SentryTransactionSamplingContext.h

```diff
@@ -0,0 +1,33 @@
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class SentryTransactionContext;
+
+NS_SWIFT_NAME(TransactionSamplingContext)
+@interface SentryTransactionSamplingContext : NSObject
+
+/**
+ * Transaction context.
+ */
+@property (nonatomic, readonly) SentryTransactionContext *transactionContext;
+
+/**
+ * Custom data used for sampling.
+ */
+@property (nonatomic, readonly) NSDictionary<NSString *, id> *customSamplingContext;
```

Comment:
`m`: In Java, this is a class, see https://github.com/getsentry/sentry-java/blob/5095cec2fc61d6129102389d2c2b8f363ac44f46/sentry/src/main/java/io/sentry/CustomSamplingContext.java#L13. Why choose we to do otherwise?

Reply:
Following .NET approach. 

@marandaneto Any thoughts on this?

Reply:
In other SDKs this is mainly a plain map instead of a type like in Java. We can discuss which way we want to go here given what we know today about trade-offs.

Reply:
Should this be a map string-string though?

Reply:
.NET uses IReadOnlyDictionary<string, object?>, that's why I didn't go will string-string

Reply:
Oh that makes sense. We can put actual objects.

Reply:
no strong feelings about that, `CustomSamplingContext` on Java is simply a wrapper around a `HashMap` that hides everything but `set` and `get`, so the question to answer is, do we want to expose every method from the `Map` contract? I don't see problems with either solutions

Reply:
Then I prefer keeping it as it is to be consistent with other parts of the SDK as `SentryOptions` or `SentryScope`.